packages/synapse-react-client/src/components/GoalsV2/GoalsV2.test.tsx

```diff
@@ -0,0 +1,88 @@
+import { render, screen, waitFor } from '@testing-library/react'
+import '@testing-library/jest-dom'
+import { server, rest } from '../../mocks/msw/server'
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+import GoalsV2 from './GoalsV2'
+import {
+  SynapseContextProvider,
+  SynapseContextType,
+} from '../../utils/context/SynapseContext'
+
+const mockSynapseContext: Partial<SynapseContextType> = {
+  isInExperimentalMode: false,
+  utcTime: false,
+  withErrorBoundary: false,
+  downloadCartPageUrl: '/DownloadCart',
+  appId: undefined,
+}
+
+const queryClient = new QueryClient()
+
+beforeAll(() => server.listen())
+afterEach(() => server.resetHandlers())
+afterAll(() => server.close())
+
+describe('GoalsV2', () => {
+  test('renders the component with provided props', async () => {
+    render(
+      <QueryClientProvider client={queryClient}>
+        <SynapseContextProvider synapseContext={mockSynapseContext}>
+          <GoalsV2
+            entityId="syn22315959"
+            dataLink="https://eliteportal.synapse.org/Explore/Data"
+          />
+        </SynapseContextProvider>
+      </QueryClientProvider>,
+    )
```

Comment:
We have a test wrapper that provides a QueryClient and SynapseContext, `createWrapper` in TestingLibraryUtils.tsx. You can do `render(<MyComponent />, { wrapper: createWrapper() })`